src/main/java/com/revlearn/team1/service/transaction/TransactionServiceImp.java

```diff
@@ -34,42 +34,43 @@ public class TransactionServiceImp implements TransactionService {
 
     @Override
     public TransactionResponseDTO checkout(TransactionRequestDTO request) throws StripeException {
-        TransactionModel item = transactionMapper.fromDTO(request);
-
         // Set the Stripe API key
         Stripe.apiKey = stripeApiKey;
 
-        // Create a Product in Stripe (optional, can be reused or stored in DB)
-        var productParams = ProductCreateParams.builder()
-                .setName(item.getName())
+        // Define product data directly within PriceData
+        var productData = SessionCreateParams.LineItem.PriceData.ProductData.builder()
+                .setName(request.name()) // Set the product name or description
+                .setDescription(request.description()) // Set the product description
                 .build();
-        var product = Product.create(productParams);
 
-        // Create a Price for the product in Stripe
-        var priceParams = PriceCreateParams.builder()
-                .setProduct(product.getId())
-                .setUnitAmount(item.getPrice()) // Price in cents (e.g., $10.00 = 1000 cents)
+        // Define PriceData with the product information
+        var priceData = SessionCreateParams.LineItem.PriceData.builder()
                 .setCurrency("usd")
+                .setUnitAmount(request.price()) // Price in cents (e.g., $10.00 = 1000 cents)
```

Comment:
as long as we communicate with frontend that this is in cents instead of dollars should be fine hahah!

Reply:
yeah, that's how Stripe/chatGPT told me to do it so haha